src/server/game/Server/Packets/MiscPackets.cpp

```diff
@@ -711,3 +711,32 @@ WorldPacket const* WorldPackets::Misc::StartTimer::Write()
 
     return &_worldPacket;
 }
+
+void WorldPackets::Misc::AdventureJournalOpenQuest::Read()
+{
+    _worldPacket >> AdventureJournalID;
+}
+
+void WorldPackets::Misc::AdventureJournalStartQuest::Read()
+{
+    _worldPacket >> QuestID;
+}
+
+void WorldPackets::Misc::AdventureJournalUpdateSuggestions::Read()
+{
+    OnLevelUp = _worldPacket.ReadBit();
+}
+
+WorldPacket const* WorldPackets::Misc::AdventureJournalDataResponse::Write()
+{
+    _worldPacket.WriteBit(OnLevelUp);
+    _worldPacket.FlushBits();
+    _worldPacket << uint32(AdventureJournalDatas.size());
+    for (auto const& dataInfo : AdventureJournalDatas)
+    {
+        _worldPacket << uint32(dataInfo.AdventureJournalID);
```

Comment:
int32, as defined in WorldPackets::Misc::AdventureJournalDataResponse

Reply:
Now it was that I understood what they were saying to me. Yes, of course I cast it with the same type of data.